src/main/java/com/hubspot/jinjava/lib/filter/SelectAttrFilter.java

```diff
@@ -110,20 +110,32 @@ protected Object applyFilter(
       }
     }
 
+    String tempValue = generateTempVariable();
+    String expression = generateTempVariable(tempValue, attr);
     ForLoop loop = ObjectIterator.getLoop(var);
     while (loop.hasNext()) {
       Object val = loop.next();
+      interpreter.getContext().put(tempValue, val);
+
+      Object attrVal = interpreter.resolveELExpression(
+        expression,
+        interpreter.getLineNumber()
+      );
 
-      Object attrVal = new Variable(
-        interpreter,
-        String.format("%s.%s", "placeholder", attr)
-      )
-      .resolve(val);
       if (acceptObjects == expTest.evaluate(attrVal, interpreter, expArgs)) {
         result.add(val);
       }
     }
+    interpreter.getContext().removeResolvedExpression(expression);
 
     return result;
   }
```

Comment:
There is an unused method parameter in this method. Should we remove it? `Map<String, Object> kwargs`